### DISCUSSION TITLE
unlock with a specific user

### DISCUSSION THREAD

**Discussion by razorback91** [jagheterfredrik/wallbox-mqtt-bridge#54] [Q&A]
Hello

I have copper SB and mqtt is working fine. I can unlock etc etc.
At home I have 2 cars and I need to get the session flagged with the proper user (associated to the car) to then manager cost.
Is there a way to specify username in the unlock command to get the session mapped to the proper user ?
So far it is flagged as default.

Thank you

Xavier

**Comment by jagheterfredrik** [comment]
It’s currently picking the first user it finds because I have a single user Wallbox. I think the cleanest might be to have an unlock button per user? 🤔

**Reply by razorback91** [reply]
yes it would be the best.  Do you think you could it or should I start to see if I can find a way by forking your code ? 
I'm not a dev pro so I start something it would not be clean and it couldn't be merged later.

**Reply by razorback91** [reply]
BTW thank you so much for what you've done. It is great !!

**Reply by jagheterfredrik** [reply]
I just remembered that the Copper SB had some weird locking issues, https://github.com/jagheterfredrik/wallbox-mqtt-bridge/commit/9eebe991f5af73333540539e61920f65fdf6a2ba. I actually don't know how to unlock for a certain user on Copper SB, only on Pulsar models.

**Reply by razorback91** [reply]
ok no bid deal. thank you. I'll check. thank you for all good job

**Reply by sincze** [reply]
Would an External RFID sticker on the Wallbox body help and kick a scenario in HA if scanned with a specific device ?

**Reply by jagheterfredrik** [reply]
Still not sure how to unlock a given user on the Wallbox on a Copper SB. Don’t own one so can’t test.

**Reply by sincze** [reply]
So if I scan the RFID with my phone... I can kick of an unlock scenario in Homeassistant.
RFID (Static) -> Phone (Dynamic) -> HA  (Automation) -> MQTT -> Wallbox Unlock. 
(Assuming you are calculating usage in HA and not in the Wallbox Cloud).
